src/Microsoft.Azure.EventHubs.Processor/ICheckpointManager.cs

```diff
@@ -49,8 +49,9 @@ public interface ICheckpointManager
         /// <summary>
         /// Update the checkpoint in the store with the offset/sequenceNumber in the provided checkpoint.
         /// </summary>
+        /// <param name="lease">Partition information against which to perform a checkpoint.</param>
         /// <param name="checkpoint">offset/sequeceNumber to update the store with.</param>
-        Task UpdateCheckpointAsync(Checkpoint checkpoint);
+        Task UpdateCheckpointAsync(Lease lease, Checkpoint checkpoint);
```

Comment:
Yes and no for .NetCore. Yes if anyone implemented their own checkpoint manager but probably not. The same change is also needed for Java. It is more dreadful there.

---
In reply to: [100396157](https://github.com/Azure/azure-event-hubs-dotnet/pull/117#discussion_r100396157) [](ancestors = 100396157)